File: app/internal/class/infrastructure/repository/json/read/mapper/mapper.go
package mapper

import (
	"github.com/julianVelandia/EDteam/SOLIDyHexagonal/ProyectoCurso/internal/class/domain"
	"github.com/julianVelandia/EDteam/SOLIDyHexagonal/ProyectoCurso/internal/class/infrastructure/repository/json/dto"
)

type Mapper struct{}

func (m Mapper) DTOClassToDomain(class dto.Class) domain.Class {
	return *domain.NewClass(
		class.ClassID,
		class.Title,
		class.CreationDate,
		class.Content,
		class.ReadTime,
	)
}

File: internal/classes/infrastructure/repository/json/read/mapper/mapper.py
from datetime import datetime

from internal.classes.domain.classes import Classes


class DTOClassesMapperRead:
    def dto_class_to_domain(self, class_obj):
        return Classes(
            class_obj.ClassID,
            class_obj.Title,
            datetime.strptime(class_obj.CreationDate, "%Y-%m-%d"),
            class_obj.Content,
            class_obj.ReadTime,
        )
